### ISSUE TITLE
Highlight open parens, not just closing

### ISSUE THREAD

**Issue by mcandre** [tsdh/highlight-parentheses.el#16]
I'm trying to emulate the appearance of Emacs' built-in parens highlighting, but when I apply:

    (setq hl-paren-colors '("firebrick1"))

and then place the cursor over the opening parenthesis, nothing is highlighted. Only the closing parenthesis highlights with the cursor over it.

**Comment by tsdh** [comment]
@mcandre I think you've observed wrongly.  When point is on the opening paren, the surrounding level of paren is highlighted, not the paren you are currently on and its closing counterpart.

And this is by design.  Note that `highlight-paren-mode` (with just one color defined) is not an extension of Emacs' `show-paren-mode`.  They behave different in two situations:

1. When point is directly on an opening paren, `show-paren-mode` highlights that and the corresponding closing paren.  In contrast, `highlight-paren-mode` highlights the pair of parens surrounding the current form where the current form is the list starting at the opening paren point is currently on.

2. When point is directly after a closing paren, `show-paren-mode` highlights that and the corresponding opening paren.  In contrast, again `highlight-paren-mode` highlights the pair of parens surrounding the preceeding form where the preceeding form is the list ending at the closing paren point is currently on.

So basically, `show-paren-mode` highlights the pair of parens when you are on a paren, and `highlight-paren-mode` highlights the parens surrounding the current form (one or many levels depending on the number of colors you have defined in `hl-paren-colors`).

`highlight-paren-mode` is designed this way because it doesn't want to supersede `show-paren-mode`, rather it is intended to be used in conjunction with `show-paren-mode`, i.e., `show-paren-mode` highlights the current pair of parens (when point is on/directly after a paren), and `highlight-paren-mode` highlights one or many levels of surrounding parens.

BTW, I think this is a duplicate of https://github.com/tsdh/highlight-parentheses.el/issues/13.

**Comment by mcandre** [comment]
Understood. Do you know of any minor modes that do supercede show-paren-mode? My use case is a minor mode equivalent.

Or, is there a way to turn on show-paren-mode when entering programming buffers, and disable it when leaving for other types of buffers?

**Comment by tsdh** [comment]
@mcandre Ah, so your only problem is that `show-paren-mode` is a global minor-mode whereas you'd like to have it enabled only in programming buffers?  Well, that's easy.  Just put the below into your `~/.emacs`:

``` elisp
(show-paren-mode -1)                          ;; turn it off globally
(make-variable-buffer-local 'show-paren-mode) ;; make it buffer-local
(add-hook 'prog-mode-hook #'show-paren-mode)  ;; enable it in programming modes
```

**Comment by mcandre** [comment]
@tsdh Thank you!!
